RecoTracker/MeasurementDet/plugins/TkPixelMeasurementDet.cc

```diff
@@ -33,8 +33,28 @@ bool TkPixelMeasurementDet::measurements( const TrajectoryStateOnSurface& stateO
     return true;
   }
   
+  // do not use this as it does not account for APE...
+  // auto xyLimits = est.maximalLocalDisplacement(stateOnThisDet,fastGeomDet().specificSurface());
+  auto le = stateOnThisDet.localError().positionError();
+  LocalError lape = static_cast<TrackerGeomDet const &>(fastGeomDet()).localAlignmentError();
+  float xl = le.xx();
+  float	yl = le.yy();
+  if (lape.valid()) {
+    xl+=lape.xx();
+    yl+=lape.yy();
+  }
+  // 5 sigma to be on the safe side
+  xl = 5.f*std::sqrt(xl);
+  yl = 5.f*std::sqrt(yl);
+
+  // do not apply for iteration not cutting on propagation
+  if (est.maxSagitta() <0 ) {
```

Comment:
You can probably squeeze cpu a bit more if you compute x1 and y1 above only  if (est.maxSagitta() >= 0 )